synapse/storage/databases/main/event_federation.py

```diff
@@ -784,6 +806,17 @@ def get_backfill_points_in_room_txn(
                      * necessarily safe to assume that it will have been completed.
                      */
                     AND edge.is_state is ? /* False */
+                    /**
+                     * We only want backwards extremities that are older than or at
+                     * the same position of the given `current_depth` (where older
+                     * means less than the given depth) because we're looking backwards
+                     * from the `current_depth` when backfilling.
+                     *
+                     *                           current_depth (ignore events that come after this, ignore 2-4)
+                     *                           |
+                     * <oldest-in-time> [0]<--[1]▼<--[2]<--[3]<--[4] <newest-in-time>
```

Comment:
given a `depth` points to a specific event... should the ascii art arrow point to `[1]` rather than between `[1]` and `[2]` ?

Reply:
hmmm, I though `depth` was the position between events because [`Tokens are positions between events`](https://github.com/matrix-org/synapse/blob/f5aaa55e2702af3cac1e195bf5d703970c24ff29/synapse/types.py#L406).

Maybe that is wrong but I think they should both align here 🤔

Reply:
hrm, well a token might be conceptually between events, but still, they implement that concept in terms of "the depth of the preceding event". Once we're talking about depth rather than tokens, I think it's fair to say that we're talking about events rather than the gaps between them.

> And multiple events can have the same depth so which event does it specifically point to?

yes, there can be multiple events with the same depth, but your ascii art shows a linear timeline so there is only one event with each depth :)

Reply:
> Once we're talking about depth rather than tokens, I think it's fair to say that we're talking about events rather than the gaps between them.

It's reasonable. Updated 👍

Still thinking about it a bit